### PR TITLE
V3 unable to list webhooks

### PR DESCRIPTION
**Pull request by Rrathinasabapath**
Updated method listWebhookNotifications to ignore options 'ProgramToken' and 'Type'. Updated unit tests missing checks for parameters in listUsers method. Please verify and provide your approval. This is for the ticket https://engineering.paypalcorp.com/jira/browse/PPPLMER-92804.

### CODE REVIEW

File: src/Hyperwallet.js [hunk 1]
```diff
@@ -1651,7 +1651,7 @@ export default class Hyperwallet {
      * @param {api-callback} callback - The callback for this call
      */
     listWebhookNotifications(options, callback) {
-        const LIST_WEBHOOK_NOTIFICATIONS_FILTERS = ["programToken", "type"];
+        const LIST_WEBHOOK_NOTIFICATIONS_FILTERS = [];
```
Comment (ramahalingam) [inline src/Hyperwallet.js:1654]: This constant can be removed.

File: src/Hyperwallet.js [hunk 2]
```diff
@@ -1651,7 +1651,7 @@ export default class Hyperwallet {
...
         if (options && !this.isValidFilter(options, LIST_WEBHOOK_NOTIFICATIONS_FILTERS)) {
             throw new Error("Invalid Filter. Expected - ".concat(LIST_WEBHOOK_NOTIFICATIONS_FILTERS));
```
Comment (ramahalingam) [inline src/Hyperwallet.js:1656]: Please remove this concatenation.

File: src/Hyperwallet.js [hunk 3]
```diff
@@ -2016,7 +2016,11 @@ export default class Hyperwallet {
      * @param {Object} inputFilters - The query parameters in the URI
```
Comment (ramahalingam) [inline src/Hyperwallet.js:2016]: Change this to options.

File: src/Hyperwallet.js [hunk 4]
```diff
@@ -2016,7 +2016,11 @@ export default class Hyperwallet {
...
      * @param {Object} listFilters - Defined list of filters for a business object
      */
-    isValidFilter(inputFilters, listFilters) {
-        return Object.keys(inputFilters).every(elem => listFilters.includes(elem));
+    isValidFilter(options, listFilters) {
+        const optionKeys = Object.keys(options);
+        if (optionKeys.length === 0) {
+            return true;
+        }
+        return listFilters.every(elem => optionKeys.includes(elem));
```
Comment (ramahalingam) [inline src/Hyperwallet.js:2024]: Think the logic here is incorrect. Please double-check.

**Review by ramahalingam — CHANGES_REQUESTED** [review]
Please remove the package-lock.json and .DS_Store file in the changes.

File: src/Hyperwallet.js (hunk 1 above)
Comment (Rrathinasabapath) [reply src/Hyperwallet.js]: Done and pushed the code correction.

File: src/Hyperwallet.js (hunk 2 above)
Comment (Rrathinasabapath) [reply src/Hyperwallet.js]: Done and pushed the code correction.

File: src/Hyperwallet.js (hunk 3 above)
Comment (Rrathinasabapath) [reply src/Hyperwallet.js]: Done.

File: src/Hyperwallet.js (hunk 4 above)
Comment (Rrathinasabapath) [reply src/Hyperwallet.js:2020]: The logic:
 Check if there are no query options in the request body:
     - if the options dont have any parameter, return that its a valid query body
     - If the options dont contain any of the mandatory fields, then return that its an invalid query body. 
    
Please let me know if the code change dont satisfy the logic mentioned above.

Comment (Rrathinasabapath) [reply src/Hyperwallet.js]: Reversed the logic as discussed and tested with data. Please review and approve.

**Review by jbausch-hw — APPROVED** [review]

**Review by arao6 — APPROVED** [review]

**Review by jbausch-hw — APPROVED** [review]

**Review by vkorotchenko — APPROVED** [review]

**Review by ramahalingam — APPROVED** [review]

**Pull request merged**